instr.ml

```diff
@@ -226,14 +229,17 @@ let required_vars = function
   | Label _l | Goto _l -> VarSet.empty
   | Comment _ -> VarSet.empty
   | Print e -> expr_vars e
-  | Osr (e, _, _, _, osr) ->
+  | Osr {cond; map} ->
     let exps = List.map (function
         | Osr_const (_, e) -> e
         | Osr_mut (_, e) -> e
         | Osr_mut_ref (_, x) -> Simple (Var x)
-        | Osr_mut_undef _ -> Simple (Constant Nil)) osr in
-    let exps_vars = List.map expr_vars exps in
-    List.fold_left VarSet.union (expr_vars e) exps_vars
+        | Osr_mut_undef _ -> Simple (Constant Nil)) map in
+    let vs1 = List.map expr_vars cond in
+    let vs2 = List.map expr_vars exps in
+    let v1 = List.fold_left VarSet.union VarSet.empty vs1 in
+    let v2 = List.fold_left VarSet.union VarSet.empty vs2 in
```

Comment:
This and the `Call` case above are calling for of a `list_expr_vars` auxiliary function.